scripts/convert_pretrained_checkpoint.py

```diff
@@ -0,0 +1,83 @@
+import json
+import shutil
+import sys
+from dataclasses import asdict
+from pathlib import Path
+
+import torch
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+from lit_gpt import Config
+from lit_gpt.utils import incremental_save, lazy_load
+
+
+@torch.inference_mode()
+def convert_checkpoint(
+    checkpoint_file: Path,
+    tokenizer_path: Path,
+    config_name: str,
+    output_folder: Path,
+    model_key: str = "model",
+) -> None:
+    """Convert a checkpoint after pretraining.
+
+    The pretrained checkpoint contains optimizer states and several other metadata that are not needed after training
+    is finished. This script will export the state-dict of the model and place it in the chosen output folder together
+    with the tokenizer and model config, which then can be loaded by other scripts for inference, evaluation, etc.
+
+    Args:
+        checkpoint_file: Path to a checkpoint file scripts produced by the scripts in `lit-gpt/pretrain/`.
+        tokenizer_path: A path to the folder that holds the tokenizer configuration files that were used to train
+            the model. All files with a name starting with 'tokenizer' will be copied to the output folder.
+        config_name: The name of the model loaded with the ``lit_gpt.Config``. The configuration will be saved as a
+            JSON file to the output folder.
+        output_folder: The output folder where model state-dict file, the tokenizer config file, and the model config
+            file will be saved.
+        model_key: The key in the checkpoint file under which the model state dict is saved.
+    """
+
+    if output_folder.is_dir() and output_folder.glob("*"):
+        raise FileExistsError(
+            f"The output folder exists and is not empty: {str(output_folder)}."
+            " Please delete it first or choose a different name."
+        )
+
+    output_folder.mkdir(parents=True)
+    output_checkpoint_file = output_folder / "lit_model.pth"
+    output_config_file = output_folder / "lit_config.json"
+
+    # Save the config to output folder
+    config = Config.from_name(config_name)
+    with open(output_config_file, "w") as json_config:
+        json.dump(asdict(config), json_config)
```

Comment:
(no need to act here)

we should implement https://github.com/Lightning-AI/lit-gpt/issues/483 which would replace this bit and be more reliable to changes done by the users in the training scripts themselves. Then instead of creating a new config, we would copy the existing one